exonum-java-proofs/src/main/java/com/exonum/binding/storage/proofs/map/flat/UncheckedFlatMapProof.java

```diff
@@ -22,8 +25,11 @@
 
   private final List<MapProofEntry> proofList;
 
-  UncheckedFlatMapProof(List<MapProofEntry> proofList) {
-    this.proofList = proofList;
+  @SuppressWarnings("unused") // Native API
+  UncheckedFlatMapProof(MapProofEntry[] proofList) {
```

Comment:
I'd better keep a constructor accepting a `List` of entries and not doing anything with them; and add a new one accepting an array of entries, sorting them, and creating an `UncheckedFlatMapProof`. E.g.:
```java
@SuppressWarnings("unused") // Native API
static UncheckedFlatMapProof fromUnsorted(MapProofEntry[] proofEntries) {
  // do your job
  return new UncheckedFlatMapProof(sortedProofEntries);
}
```
or, if you prefer inlining, you may add a method that sorts an array and returns it as a list, and just call
```
this(sort(proofEntries)));
```